src/Articulate/UmbracoHelperExtensions.cs

```diff
@@ -406,8 +407,81 @@ private static IPublishedContent[] GetListNodes(IMasterModel masterModel)
             }
             return listNodes;
         }
-        
-        
+
+        /// <summary>
+        /// Get all recent blog articles sorted
+        /// </summary>
+        /// <param name="helper"></param>
+        /// <param name="id"></param>
+        /// <param name="sortType"></param>
+        /// <returns>IEnumerable<PostModel></returns>
+        public static IEnumerable<PostModel> GetSortedAllPosts(UmbracoHelper helper, int id, string sortType)
+        {           
+            IContentService contentService = Current.Services.ContentService;
+
+            var publishedContent = new MasterModel(helper.Content(id));
+
+            var listNodes = publishedContent.RootBlogNode.Children("ArticulateArchive").ToArray();
+
+            var listNodeIds = listNodes.Select(x => x.Id).ToArray();
+
+            var totalPosts = GetPostCount(helper, listNodes.Select(x => x.Id).ToArray());
+
+            var pager = new PagerModel(totalPosts, 0, 1);
+
+            var listItems = helper.GetPostsSortedByPublishedDate(pager, null, listNodeIds);
```

Comment:
This should only be used when sorting by published date, else this is also causing a lot of overhead to resolve the nodes, sort them and then resort them again below if the sorting has changed. Probably best to add other ext methods for UmbracoHelper for the specific sorting required like GetPostsSortedBy...